core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -201,32 +215,42 @@ private ClientRWLock getBlockLock(long blockId) {
     // Loop until we either find the block lock in the mLocks map, or successfully acquire a new
     // block lock from the lock pool.
     while (true) {
-      ClientRWLock blockLock;
       // Check whether a lock has already been allocated for the block id.
-      try (LockResource r = new LockResource(mSharedMapsLock.readLock())) {
-        blockLock = mLocks.get(blockId);
-        if (blockLock != null) {
-          blockLock.addReference();
-          return blockLock;
-        }
+      ClientRWLock reuseExistingLock = mLocks.computeIfPresent(
+          blockId,
+          (blkid, lock) -> {
+            lock.addReference();
+            return lock;
+          }
+      );
+      if (reuseExistingLock != null) {
+        return reuseExistingLock;
       }
       // Since a block lock hasn't already been allocated, try to acquire a new one from the pool.
-      // Acquire the lock outside the synchronized section because #acquire might need to block.
       // We shouldn't wait indefinitely in acquire because the another lock for this block could be
       // allocated to another thread, in which case we could just use that lock.
-      blockLock = mLockPool.acquire(1, TimeUnit.SECONDS);
-      if (blockLock != null) {
-        try (LockResource r = new LockResource(mSharedMapsLock.writeLock())) {
+      ClientRWLock newlyAcquiredLock = mLockPool.acquire(1, TimeUnit.SECONDS);
+      if (newlyAcquiredLock != null) {
+        int referenceCount = newlyAcquiredLock.getReferenceCount();
+        if (referenceCount != 0) {
+          LOG.warn("A block lock was not cleanly released as newly acquired locks should have 0 "
```

Comment:
it's better to use LOG.error if we don't want to throw exceptions when acquiring locks.